components/base/kinematicbase/ptgKinematics.go

```diff
@@ -146,7 +146,9 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 	selectedPTG := ptgk.ptgs[int(math.Round(inputs[ptgIndex].Value))]
 	selectedTraj, err := selectedPTG.Trajectory(inputs[trajectoryIndexWithinPTG].Value, inputs[distanceAlongTrajectoryIndex].Value)
 	if err != nil {
-		return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
+		stopCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancelFn()
+		return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
```

Comment:
If the context is already cancelled, the original code would fail to stop the robot.